Now it is

```
>>> N = ReferenceFrame('N')
>>> A = ReferenceFrame('A')
>>> theta = dynamicsymbols('theta')
>>> A.orient_axis(N, theta, N.x + N.y + N.z)

Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/root/.vscode-server/extensions/ms-python.python-2021.6.944021595/pythonFiles/lib/python/debugpy/__main__.py", line 45, in <module>
    cli.main()
  File "/root/.vscode-server/extensions/ms-python.python-2021.6.944021595/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 444, in main
    run()
  File "/root/.vscode-server/extensions/ms-python.python-2021.6.944021595/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 285, in run_file
    runpy.run_path(target_as_str, run_name=compat.force_str("__main__"))
  File "/usr/lib/python3.8/runpy.py", line 265, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/usr/lib/python3.8/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/root/sympy/sympy/physics/vector/tests/test_frame.py", line 556, in <module>
    A.orient_axis(N, theta, N.x + N.y + N.z)
  File "/root/sympy/sympy/physics/vector/frame.py", line 649, in orient_axis
    axis = _check_vector(axis)
  File "/root/sympy/sympy/physics/vector/vector.py", line 807, in _check_vector
    raise TypeError('A Vector must be supplied')
TypeError: A Vector must be supplied
```

--- a/sympy/physics/vector/frame.py
+++ b/sympy/physics/vector/frame.py
@@ -647,8 +647,7 @@ class ReferenceFrame:
         _check_frame(parent)
 
         if not isinstance(axis, Vector) and isinstance(angle, Vector):
-            raise TypeError('The order of arguments is incorrect.' +\
-                ' The arguments are orient_axis(frame, axis, angle)')
+            axis, angle = angle, axis
 
         axis = _check_vector(axis)
         amount = sympify(angle)

--- a/sympy/physics/vector/frame.py
+++ b/sympy/physics/vector/frame.py
@@ -646,6 +646,10 @@ class ReferenceFrame:
         from sympy.physics.vector.functions import dynamicsymbols
         _check_frame(parent)
 
+        if not isinstance(axis, Vector) and isinstance(angle, Vector):
+            raise TypeError('The order of arguments is incorrect.' +\
+                ' The arguments are orient_axis(frame, axis, angle)')
+
         axis = _check_vector(axis)
         amount = sympify(angle)
         theta = amount

--- a/sympy/physics/vector/tests/test_frame.py
+++ b/sympy/physics/vector/tests/test_frame.py
@@ -454,6 +454,8 @@ def test_orient_axis():
     A = ReferenceFrame('A')
     B = ReferenceFrame('B')
     assert A.orient_axis(B,-B.x, 1) == A.orient_axis(B, B.x, -1)
+    raises(TypeError, lambda: A.orient_axis(B, 1, -B.x))
+    raises(TypeError, lambda: A.orient_axis(B, 1, 1))
 
 def test_orient_body():
     A = ReferenceFrame('A')

--- a/sympy/physics/vector/tests/test_frame.py
+++ b/sympy/physics/vector/tests/test_frame.py
@@ -454,7 +454,7 @@ def test_orient_axis():
     A = ReferenceFrame('A')
     B = ReferenceFrame('B')
     assert A.orient_axis(B,-B.x, 1) == A.orient_axis(B, B.x, -1)
-    raises(TypeError, lambda: A.orient_axis(B, 1, -B.x))
+    assert A.orient_axis(B,-B.x, 1) == A.orient_axis(B, 1, -B.x)
     raises(TypeError, lambda: A.orient_axis(B, 1, 1))
 
 def test_orient_body():

